lib/config/migrations/migration.ts

```diff
@@ -0,0 +1,18 @@
+import type { RenovateConfig } from '../types';
+
+export abstract class Migration {
+  protected readonly originalConfig: RenovateConfig;
+
+  protected readonly migratedConfig: RenovateConfig;
+
+  constructor(originalConfig: RenovateConfig, migratedConfig: RenovateConfig) {
+    this.originalConfig = originalConfig;
+    this.migratedConfig = migratedConfig;
+  }
+
+  abstract migrate(): RenovateConfig;
```

Comment:
```suggestion
  abstract migrate(): void;
```
We do not need / use the return value.

Reply:
Done